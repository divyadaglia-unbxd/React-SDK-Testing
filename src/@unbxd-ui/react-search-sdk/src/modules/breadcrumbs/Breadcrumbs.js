import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import BreadcrumbsContainer from './BreadcrumbsContainer';

/**
 * Component to manage the breadcrumb.
 */
const Breadcrumbs = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(Breadcrumbs.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    productType
                } = appState;

                return (
                    <BreadcrumbsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        productType={productType}
                        helpers={helpers}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

Breadcrumbs.displayName = 'Breadcrumbs';

Breadcrumbs.defaultProps = {};

Breadcrumbs.propTypes = {
    /**
     * root of the breadcrumb.
     */
    root: PropTypes.oneOfType([PropTypes.element, PropTypes.node]),
    /**
     * Separator of the breadcrumb.
     */
    separator: PropTypes.node,
    /**
     * Custom breadcrumb component instance.
     */
    breadcrumbItemComponent: PropTypes.element,
    /**
     * Callback for breadcrumb click.
     */
    onBreadcrumbClick: PropTypes.func
};

export default Breadcrumbs;
