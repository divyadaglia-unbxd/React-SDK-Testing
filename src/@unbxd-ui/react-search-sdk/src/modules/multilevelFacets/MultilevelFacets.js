import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import MultilevelFacetsContainer from './MultilevelFacetsContainer';

/**
 * Component to manage multilevel categories.
 */
const MultilevelFacets = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(MultilevelFacets.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    productType
                } = appState;

                return (
                    <MultilevelFacetsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        productType={productType}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

MultilevelFacets.displayName = 'MultilevelFacets';

MultilevelFacets.defaultProps = {
    collapsible: false,
    searchable: false,
    minViewMore: 3,
    enableViewMore: false
};

MultilevelFacets.propTypes = {
    /**
     * Collapse facet values
     */
    collapsible: PropTypes.bool,
    /**
     * Search facet values
     */
    searchable: PropTypes.bool,
    /**
     * Min value for viewMore to be enabled.
     */
    minViewMore: PropTypes.number,
    /**
     * Bool value to enable/disable viewMore.
     */
    enableViewMore: PropTypes.bool,
    /**
     * Custom facet element
     */
    facetItemComponent: PropTypes.element,
    /**
     * Label for the module.
     */
    label: PropTypes.node,
    /**
     * Callback for facet change.
     */
    onFacetClick: PropTypes.func
};

export default MultilevelFacets;
