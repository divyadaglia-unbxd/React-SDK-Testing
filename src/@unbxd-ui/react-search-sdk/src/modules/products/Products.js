import React from 'react';
import PropTypes from 'prop-types';

import ProductsContainer from './ProductsContainer';
import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';

/**
 * Component to manage products.
 * Products also manages the pagination options for the search results here.
 */
const Products = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(Products.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    unbxdState,
                    priceUnit
                } = appState;
                const { viewType, pageSize, sort } = unbxdState;

                return (
                    <ProductsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        viewType={viewType}
                        pageSize={pageSize}
                        sort={sort}
                        priceUnit={priceUnit}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

Products.displayName = 'Products';

Products.defaultProps = {
    perRow: 4,
    pageSize: 10,
    variantsCount: 0,
    paginationType: 'FIXED_PAGINATION',
    heightDiffToTriggerNextPage: 300,
    showLoader: false,
    productIdAttribute: 'uniqueId',
    groupBy: '',
    variantAttributesMap: {},
    onResultsLoaded:()=>{}
};

Products.propTypes = {
    /**
     * Number of products to be shown per row.
     */
    perRow: PropTypes.number,
    /**
     * Required pagination type. Possible options are `INFINITE_SCROLL`, `CLICK_N_SCROLL` and `FIXED_PAGINATION`.
     */
    paginationType: PropTypes.string,
    /**
     * Height difference to trigger for next page in case of paginationType `INFINITE_SCROLL`.
     */
    heightDiffToTriggerNextPage: PropTypes.number,
    /**
     * Custom load more component instance
     */
    loadMoreComponent: PropTypes.element,
    /**
     * Unique attribute of the product
     */
    productIdAttribute: PropTypes.string,
    /**
     * Mapping of catalog Product fields to SDK Product fields.
     */
    attributesMap: PropTypes.object.isRequired,
    /**
     * Show variants to a product.
     */
    showVariants: PropTypes.bool,
    /**
     * Number of variants to fetch.
     */
    variantsCount: PropTypes.number,
    /**
     * Mapping of catalog Product variant fields to SDK Product variant fields.
     */
    variantAttributesMap: PropTypes.object,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
     * Custom loader component instance
     */
    loaderComponent: PropTypes.element,
    /**
     * Callback function triggered on click of a product.
     */
    onProductClick: PropTypes.func,
    /**
     * Callback function triggered on zero results.
     */
    onZeroResults: PropTypes.func,
    /**
     * Display swatches
     */
    showSwatches: PropTypes.bool,
    /**
     * Group variants by the attribute
     */
    groupBy: PropTypes.string,
    /**
     * Swatch attributes that change on click on of the swatch
     */
    swatchAttributesMap: PropTypes.object,
    /**
     * Custom swatch component instance
     */
    swatchItemComponent: PropTypes.element,
    /**
     * Custom product item component instance
     */
    productItemComponent: PropTypes.element,
    /**
     *  Custom zero result component instance
     */
    zeroResultsComponent: PropTypes.element,
    /**
     *  callback on each new results load
     */
     onResultsLoaded: PropTypes.func
};

export default Products;
