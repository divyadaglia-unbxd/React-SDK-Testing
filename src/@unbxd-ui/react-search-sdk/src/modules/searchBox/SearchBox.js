import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import SearchBoxContainer from './SearchBoxContainer';

/**
 * Component to display merchandising banners.
 */
const SearchBox = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(SearchBox.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    productType,
                    unbxdState
                } = appState;

                const { query } = unbxdState;

                return (
                    <SearchBoxContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        productType={productType}
                        query={query}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

SearchBox.displayName = 'SearchBox';

SearchBox.defaultProps = {
    autoFocus: false,
    clearable: false,
    showLoader: false
};

SearchBox.propTypes = {
    /**
     * Should the searchbox be focused by default.
     */
    autoFocus: PropTypes.bool,
    /**
     * Should the searchbox be clearable.
     */
    clearable: PropTypes.bool,
    /**
     * Hook for search query. The function should return `true` if the search is to be triggered, false otherwise.
     */
    onSubmit: PropTypes.func,
    /**
     * Hook for clearing the search query. The function should return `true` if the searchbox is to be cleared, false otherwise.
     */
    onClear: PropTypes.func,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
     * Custom input component instance
     */
    inputComponent: PropTypes.element,
    /**
     * Custom submit component instance
     */
    submitComponent: PropTypes.element,
    /**
     * Custom reset component instance
     */
    clearComponent: PropTypes.element,
    /**
     * Default search query
     */
    defaultSearch: PropTypes.string,
    /**
     * SearchBox placeholder
     */
    placeholder: PropTypes.string
};

export default SearchBox;
