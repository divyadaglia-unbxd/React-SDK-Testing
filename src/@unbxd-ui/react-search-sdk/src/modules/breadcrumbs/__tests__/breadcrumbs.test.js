import renderer from 'react-test-renderer';
import {
    render,
    waitFor,
    fireEvent,
    screen,
    getByTestId
} from '@testing-library/react';
import React from 'react';
import Products from '../../products/';
import Breadcrumbs from '../index';
import MultilevelFacets from '../../multilevelFacets';
import UnbxdSearchWrapper from '../../../UnbxdSearchWrapper';
import SearchBox from '../../searchBox';
import { breadcrumbSearchResponse, setsBreadcrumbResponse } from './mocks/';

// establish API mocking before all tests
beforeAll(() => {
    window.fetch = jest.fn((request) => {
        if (request.includes('filter=SUBCATEGORY:"Sets"')) {
            return Promise.resolve({
                json: () => Promise.resolve(setsBreadcrumbResponse)
            });
        }
        return Promise.resolve({
            json: () => Promise.resolve(breadcrumbSearchResponse)
        });
    });
});

const attributesMap = {
    title: 'title',
    uniqueId: 'uniqueId',
    imageUrl: 'imageUrl',
    price: 'RRP_Price',
    productUrl: 'productUrl'
};

const Root = () => <span className="UNX-breadcrumb__root">Home</span>;
const separator = <span className="UNX-breadcrumb__separator">/</span>;

test('Match Snapshot for breadcrumbs', async () => {
    const tree = renderer.create(
        <UnbxdSearchWrapper
            siteKey="wildearthclone-neto-com-au808941566310465"
            apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
        >
            <Breadcrumbs root={<Root />} separator={separator} />
            <Products attributesMap={attributesMap} />
            <div>
                <SearchBox defaultSearch="shoes" />
            </div>
        </UnbxdSearchWrapper>
    );
    await waitFor(() => tree.toJSON());
    expect(tree.toJSON()).toMatchSnapshot();
});

test('Breadcrumbs test', async () => {
    const { getByText, getAllByText, container } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <Breadcrumbs root={<Root />} separator={separator} />
                <MultilevelFacets />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(getByText('SUBCATEGORY')).toBeInTheDocument();
        fireEvent.click(getByText('Sets'));
    });
    await waitFor(() => {
        getAllByText('Home');
        fireEvent.click(getAllByText('Sets')[0]);
    });
});

test('Breadcrumbs Item test', async () => {
    const BreadcrumbItemComponent = ({ itemData, idx, onClick }) => {
        const { value } = itemData;
        const handleClick = () => {
            onClick(itemData);
        };
        return (
            <>
                {idx === 0 && <Root />}
                {separator}
                <div
                    className="UNX-breadcrumbs-list-item"
                    data-testid="UNX-breadcrumbs-list-item-custom"
                    onClick={handleClick}
                >
                    {value}
                </div>
            </>
        );
    };

    const { getByText, getAllByTestId } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <Breadcrumbs
                    breadcrumbItemComponent={<BreadcrumbItemComponent />}
                />
                <MultilevelFacets />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(getByText('Sets')).toBeInTheDocument();
        fireEvent.click(getByText('Sets'));
    });

    await waitFor(() => {
        expect(getAllByTestId('UNX-breadcrumbs-list-item-custom'));
    });
});
