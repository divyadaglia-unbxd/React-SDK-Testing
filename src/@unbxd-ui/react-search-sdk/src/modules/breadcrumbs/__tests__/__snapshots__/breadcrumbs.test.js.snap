// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot for breadcrumbs 1`] = `
Array [
  <div
    className="UNX-breadcrumbs__container"
  >
    <div
      className="UNX-breadcrumbs__list"
    />
  </div>,
  <div
    className="UNX-breadcrumbs__container"
  >
    <div
      className="UNX-breadcrumbs__list"
    />
  </div>,
  <div>
    <div
      className="UNX-products__container"
    >
      <div
        className="UNX-products__list -grid grid-cols-4"
      >
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            onClick={[Function]}
          >
            <img
              className="-image"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            />
            <div
              className="-price"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="UNX-searchbox__container"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="UNX-searchbox__input"
          onChange={[Function]}
          placeholder=""
          value="*"
        />
        <button
          aria-label="search"
          className="UNX-searchbox__button"
          tabIndex={0}
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>,
]
`;
